artiq/coredevice/mirny.py

```diff
@@ -140,3 +151,143 @@ def write_ext(self, addr, length, data):
         if length < 32:
             data <<= 32 - length
         self.bus.write(data)
+
+
+class Almazny:
+    """
+    Almazny (High frequency mezzanine board for Mirny)
+
+    :param host_mirny - Mirny device Almazny is connected to
+    """
+
+    def __init__(self, dmgr, host_mirny):
+        self.mirny = dmgr.get(host_mirny)
+        self.att_mu = [0x3f] * 4
+        self.rf_switches = [0] * 4
+
+    @kernel
+    def init(self):
+        self._send_mezz_data(ALMAZNY_REG_CLEAR)  # reset the rest
```

Comment:
Is this necessary?
My understanding is that this simply resets the data preloaded in the shift registers (left columns of FFs) which you are systematically overwriting later before pulsing the latch signal that transfers it to the right columns of FFs and the outputs. If this understanding is correct, this reset can and should be removed entirely, unless you also wanted to latch afterwards. But in this case, it is simpler to just load the values of ``att_mu`` and ``rf_switches`` in ``init`` so that there is a single source of truth for the initial values of the attenuators and RF switches.
![image](https://user-images.githubusercontent.com/720864/142146281-a1f8f8d1-3b27-461f-bf0d-db2a9129e3b4.png)